qiskit/transpiler/passes/synthesis/unitary_synthesis.py

```diff
@@ -488,7 +578,10 @@ def _synth_natural_direction(
             basis_fidelity = approximation_degree
         else:
             basis_fidelity = physical_gate_fidelity
-        synth_circ = decomposer2q(su4_mat, basis_fidelity=basis_fidelity)
+        if not isinstance(decomposer2q, XXDecomposer):
+            synth_circ = decomposer2q(su4_mat, basis_fidelity=basis_fidelity)
+        else:
+            synth_circ = decomposer2q(su4_mat)
```

Comment:
Needing to dispatch on the particular class implementing the decomposer _feels_ wrong - shouldn't they all be implementing the same interface?  Within that uncertainty, this check kind of feels the wrong way round - given that `XXDecomposer` can't take `basis_fidelity`, can we be sure that _all_ other decomposers can?

Reply:
Well there aren't that many, I think it's just the two here and there isn't a standard interface definition between the classes. I was surprised `XXDecomposer` didn't take a `basis_fidelity` kwarg, nothing actually ever used it before which is why I had to do this here. Adding the support for the fixed angle RZXGates enabled this and that's when I found the issue.